src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenDeconstructTests.cs

```diff
@@ -4594,5 +4594,200 @@ class var
+        }
+
+        [Fact]
+        public void ShortDiscardInDeclarationWithExistingUnderscoreLocal()
+        {
+            var source =
+@"
+class C
+{
+    static void Main()
+    {
+        int _;
+        (_, var x) = (1, 2);
+        System.Console.Write(_);
+    }
+}
+";
+
+            var comp = CreateCompilationWithMscorlib(source, options: TestOptions.DebugExe, references: s_valueTupleRefs);
+            // TODO REVIEW error for now, until we do mixing
+            //comp.VerifyDiagnostics();
+            //CompileAndVerify(comp, expectedOutput: "1");
+        }
+
+        [Fact]
+        public void DiscardInDeconstructAssignment()
+        {
+            var source =
+@"
+class C
+{
+    static void Main()
+    {
+        int x;
+        (_, x) = (1, 2);
+        System.Console.Write(x);
+    }
+}
+";
+
+            var comp = CreateCompilationWithMscorlib(source, options: TestOptions.DebugExe, references: s_valueTupleRefs);
+            // TODO REVIEW This is an error, but should be allowed once we do mixing
+            // (7,10): error CS0103: The name '_' does not exist in the current context
+            //         (_, x) = (1, 2);
+            //Diagnostic(ErrorCode.ERR_NameNotInContext, "_").WithArguments("_").WithLocation(7, 10)
+            //comp.VerifyDiagnostics();
+            //CompileAndVerify(comp, expectedOutput: "2");
+        }
+
+        [Fact]
+        public void DiscardInDeconstructAssignmentWithUnderscoreLocal()
+        {
+            var source =
+@"
+class C
+{
+    static void Main()
+    {
+        int x, _;
+        (_, x) = (1, 2);
+        System.Console.Write($""{_} {x}"");
+    }
+}
+";
+
+            var comp = CreateCompilationWithMscorlib(source, options: TestOptions.DebugExe, references: s_valueTupleRefs);
+            comp.VerifyDiagnostics();
+            CompileAndVerify(comp, expectedOutput: "1 2");
+        }
+
+        [Fact]
+        public void DeconstructAssignmentWithDiscardsOnly()
+        {
+            var source =
+@"
+class C
+{
+    static void Main()
+    {
+        (_, _) = (new C(), new C());
+    }
+    public C() { System.Console.Write(""C""); }
+}
+";
+
+            var comp = CreateCompilationWithMscorlib(source, options: TestOptions.DebugExe, references: s_valueTupleRefs);
+            // TOOD REVIEW This will become allowed
+            //comp.VerifyDiagnostics();
+            // (6,10): error CS0103: The name '_' does not exist in the current context
+            //         (_, _) = (new C(), new C());
+            //Diagnostic(ErrorCode.ERR_NameNotInContext, "_").WithArguments("_").WithLocation(6, 10),
+            // (6,13): error CS0103: The name '_' does not exist in the current context
+            //         (_, _) = (new C(), new C());
+            //Diagnostic(ErrorCode.ERR_NameNotInContext, "_").WithArguments("_").WithLocation(6, 13)
+            //CompileAndVerify(comp, expectedOutput: "CC");
+        }
+
+        // nested cases
```

Comment:
I see what you're doing, but you proabaly don't want to leave the remains of the todo list here.